ui/Makefile

```diff
@@ -16,142 +16,101 @@
+GOBINDATA_SOURCES = $(JSPM_TARGET) assets fonts favicon.ico apple-touch-icon.png index.html
 
-typings.installed: typings.json npm.installed
-	rm -rf $(TYPINGS)/
-	typings install
-	touch $@
-
-$(TYPESCRIPT_TARGET): $(shell find $(TS_ROOT)) $(REMOTE_DEPS)
-	tsc -p $(TS_ROOT)
+$(GOBINDATA_TARGET): $(GO_BINDATA) $(GOBINDATA_SOURCES)
+	$(GO_BINDATA) -nometadata -pkg ui -o $@ $(GOBINDATA_SOURCES)
+	# Add comment recognized by reviewable.
+	echo '// GENERATED FILE DO NOT EDIT' >> $@
+	gofmt -s -w $@
+	goimports -w $@
 
-$(TEST_TARGET): $(shell find $(TS_ROOT)) $(REMOTE_DEPS)
-	tsc -p $(TEST_ROOT)
+$(JSPM_TARGET): jspm.installed typings.installed $(CSS_TARGET) $(shell find $(TS_ROOT) -name '*.tsx' -or -name '*.ts')
+	jspm bundle-sfx app/app build/app.js
 
-$(CSS_TARGET): $(shell find $(STYL_ROOT)) npm.installed bower.installed
+# TODO(tamird/matt/max): write a systemjs stylus plugin.
+$(CSS_TARGET): $(shell find $(STYL_ROOT)) npm.installed
 	stylus $(STYL_ROOT)/app.styl \
-	       --compress \
 	       --include-css \
 	       --use nib \
 	       --out $@
 
-$(CSS_DEBUG_TARGET): $(shell find $(STYL_ROOT)) npm.installed bower.installed
-	stylus $(STYL_ROOT)/app.styl \
-	       --include-css \
-	       --use nib \
-	       --out $@
+npm.installed: package.json npm-shrinkwrap.json
+	rm -rf $(NODE_MODULES)/ $(TYPINGS)/
+	npm install --no-progress
+	touch $@
 
-$(GOBINDATA_TARGET): $(GOBINDATA_DEPS) bower.installed release/$(INDEX)
-	rm -f $(INDEX)
-	cp release/$(INDEX) $(INDEX)
-	chmod a-w $(INDEX)
-	$(GO_BINDATA) -nometadata -pkg ui -o $@ $(GOBINDATA_DEPS) $(INDEX) $(GOBINDATA_SOURCES)
-	# Add comment recognized by reviewable.
-	echo '// GENERATED FILE DO NOT EDIT' >> $@
-	gofmt -s -w $@
-	goimports -w $@
+typings.installed: generated/protos.d.ts typings.json npm.installed
+	typings install
+	typings prune
+	touch $@
+
+jspm.installed: npm.installed
+	jspm install
+	# jspm touches package.json, so we need to bump npm.installed's timestamp
+	# here to prevent endless pointless rebuilds. This `touch` will still result
+	# in a spurious `typings install`, but that's extremely fast compared to `npm
+	# install`.
+	touch npm.installed
+	touch $@
+
+UI_PROTOS = $(REPO_ROOT)/server/serverpb/admin.proto $(REPO_ROOT)/server/serverpb/status.proto $(REPO_ROOT)/server/status/status.proto $(REPO_ROOT)/ts/tspb/timeseries.proto
 
-$(GOBINDATA_DEBUG_TARGET): $(GOBINDATA_DEBUG_DEPS) bower.installed debug/$(INDEX)
-	rm -f $(INDEX)
-	cp debug/$(INDEX) $(INDEX)
-	chmod a-w $(INDEX)
-	$(GO_BINDATA) -nometadata -pkg ui -o $@ -debug $(GOBINDATA_DEBUG_DEPS) $(INDEX) $(GOBINDATA_DEBUG_SOURCES)
+PBJS_ARGS = --path $(ORG_ROOT) $(UI_PROTOS)
+
+app/js/protos.js: npm.installed $(UI_PROTOS)
 	# Add comment recognized by reviewable.
-	echo '// GENERATED FILE DO NOT EDIT' >> $@
+	echo '// GENERATED FILE DO NOT EDIT' > $@
+	pbjs -t commonjs $(PBJS_ARGS) >> $@
 
-watch: debug
-	gulp watch
+generated/protos.json: npm.installed $(UI_PROTOS)
+	pbjs $(PBJS_ARGS) > $@
 
-.PHONY: clean
```

Comment:
Yeah, it was intentional; it should never be necessary to use that target. I assume you ran into this because you ran the top level clean - mind removing that reference?